src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -643,13 +648,14 @@ export default class FilterOptionsComponent extends Component {
 
   /**
    * Saves selected options to persistent storage
+   * @param {boolean} replaceHistory Whether we replace or push a new history
+   *                                 state for the associated changes
    */
-  saveSelectedToPersistentStorage () {
-    const replaceHistory = (this.core.persistentStorage.get(this.name) === null);
+  saveSelectedToPersistentStorage (replaceHistory) {
     this.core.persistentStorage.set(
       this.name,
       this.config.options.filter(o => o.selected).map(o => o.label),
-      replaceHistory
+      replaceHistory || (this.core.persistentStorage.get(this.name) === null)
```

Comment:
Could we run into a situation here where `replaceHistory` is false and we overwrite it here? Should we be checking that `replaceHistory` is `undefined` instead?

Reply:
So previously, it was always that that value `(this.core.persistentStorage.get(this.name) === null)`. With these changes, we have made it so that it can be overrided to `true` (but never overrided to `false` because of the `||`) based on the `replaceHistory` parameter.

So I guess semantically, this `replaceHistory` parameter is more `overrideReplaceHistoryIfTrue`. If you wanted to check for `undefined` before defaulting, I could remove the `false` from the other function calls and change this logic to

```
(replaceHistory !== undefined) ? replaceHistory : (this.core.persistentStorage.get(this.name) === null)
```